models/migrations/v18.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018 The Gogs Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+
+	"github.com/go-xorm/xorm"
+
+	"github.com/gogs/gogs/pkg/setting"
+)
+
+func updateRepositoryDescriptionField(x *xorm.Engine) error {
+	exist, err := x.IsTableExist("repository")
+	if err != nil {
+		return fmt.Errorf("IsTableExist: %v", err)
+	} else if !exist {
+		return nil
+	}
+	switch {
+		case setting.UseMySQL:
+			_, err = x.Exec("ALTER TABLE `repository` MODIFY `description` VARCHAR(512);")
+		case setting.UseMSSQL:
+			_, err = x.Exec("ALTER TABLE `repository` ALTER COLUMN `description` VARCHAR(512);")
+		case setting.UsePostgreSQL:
+			_, err = x.Exec("ALTER TABLE `repository` ALTER COLUMN `description` TYPE VARCHAR(512);")
+		case setting.UseSQLite3:
+			// Sqlite3 uses TEXT field by default for any string type field.
+		default:
```

Comment:
In general, we would return error if not matched to known list, but since it is migration and we know what we have, `default` is not needed at all. What do you think?

Reply:
Yes, it's useless here. We check all available options here.